tests/test-st2-invalid-auth.js

```diff
@@ -0,0 +1,154 @@
+/*
+ Licensed to the StackStorm, Inc ('StackStorm') under one or more
+ contributor license agreements.  See the NOTICE file distributed with
+ this work for additional information regarding copyright ownership.
+ The ASF licenses this file to You under the Apache License, Version 2.0
+ (the "License"); you may not use this file except in compliance with
+ the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*jshint quotmark:false*/
+/*jshint -W030*/
+/*global describe, it, before, after*/
+'use strict';
+
+var chai = require("chai"),
+  expect = chai.expect,
+  chaiString = require("chai-string"),
+  sinon = require('sinon'),
+  sinonChai = require('sinon-chai'),
+  mockedEnv = require('mocked-env'),
+  Robot = require("hubot/src/robot"),
+  Logger = require('./dummy-logger.js');
+
+chai.use(sinonChai);
+chai.use(chaiString);
+
+global.process.exit = sinon.spy();
+
+describe("invalid st2 credential configuration", function() {
+  var robot = new Robot(null, "mock-adapter", false, "Hubot");
+  robot.logger = new Logger(true);
+  var restore_env = null,
+    info_spy = sinon.spy(robot.logger, 'info'),
+    error_spy = sinon.spy(robot.logger, 'error');
+
+  beforeEach(function () {
+    process.exit.resetHistory();
+  });
+
+  afterEach(function() {
+    restore_env && restore_env();
+    error_spy.resetHistory();
+    info_spy.resetHistory();
+    // Remove stackstorm.js from the require cache
+    // https://medium.com/@gattermeier/invalidate-node-js-require-cache-c2989af8f8b0
+    delete require.cache[require.resolve("../scripts/stackstorm.js")];
+  });
+
+  after(function () {
+    process.exit.restore && process.exit.restore();
+  });
+
+  it("should raise exception with null auth URL", function(done) {
+    // Mock process.env for all modules
+    // https://glebbahmutov.com/blog/mocking-process-env/
+    restore_env = mockedEnv({
+      ST2_AUTH_USERNAME: 'nonexistent-st2-auth-username',
+      ST2_AUTH_PASSWORD: 'nonexistent-st2-auth-password'
+    });
+
+    // Load script under test
+    var stackstorm = require("../scripts/stackstorm.js");
+    stackstorm(robot).then(function (stop) {
+      expect(error_spy).to.have.been.calledOnce;
+      expect(error_spy.firstCall.args[0]).include('Environment variables ST2_AUTH_USERNAME, ST2_AUTH_PASSWORD and ST2_AUTH_URL should only be used together.');
```

Comment:
@blag I think some of these tests would be very useful if extracted as separated PR. They really look nice :+1: 

It feels that we try to put too much into 1 single PR: bugfix, refactoring and significant change of logic. Would be great to stick with several independent and controlled & predictable changes.

Reply:
This is a fair criticism. It is difficult to work on code without refactoring it to be better, but I'll try to keep PRs smaller and more constrained in the future.

I do think including tests alongside the fix is appropriate though.